epowcore/pandapower/pandapower_model.py

```diff
@@ -0,0 +1,447 @@
+        slack = False
+        if synchronous_machine_bus.lf_bus_type == LFBusType("SLACK"):
+            slack = True
+            Logger.log_to_selected(
+                "Gen:" + synchronous_machine.name + " is set to be a slack genenerator"
+            )
+        # Create generator in pandapower network
+        pandapower.create_gen(
+            net=self.network,
+            name=synchronous_machine.name,
+            index=synchronous_machine.uid,
+            bus=synchronous_machine_bus.uid,
+            # active_power or rated_active_power
+            # -> makes no difference for IEEE39
+            # leaving active_power because its by definition more fitting
+            # (even though rated_active_power is used in pandapower converter)
+            p_mw=synchronous_machine.rated_active_power,
+            vm_pu=synchronous_machine.voltage_set_point,
+            # or rated active power
+            sn_mva=synchronous_machine.rated_apparent_power,
+            max_q_mvar=synchronous_machine.q_max,
+            min_q_mvar=synchronous_machine.q_min,
+            min_p_mw=synchronous_machine.p_min,
+            max_p_mw=synchronous_machine.p_max,
+            min_vm_pu=nan,
+            max_vm_pu=nan,
+            scaling=1.0,
+            type="sync",
+            slack=slack,
+            controllable=False,
+            vn_kv=synchronous_machine.rated_voltage,
+            xdss_pu=synchronous_machine.subtransient_reactance_x,
+            rdss_ohm=nan,
+            cos_phi=nan,
+            pg_percent=nan,
+            power_station_trafo=nan,
+            in_service=True,
+            slack_weight=0.0,
+        )
+        return True
+
+    def create_line_from_gdf_tline(self, core_model: CoreModel, tline: TLine):
+        """Create a pandapower line in the network equivalent to a gdf transmission line."""
+        # Get the neigbours of the transmission line to know what it connects to
+        from_bus = core_model.get_neighbors(component=tline, follow_links=True, connector="A")[0]
+        to_bus = core_model.get_neighbors(component=tline, follow_links=True, connector="B")[0]
+        # Conversion fails if one of the buses isn't found
+        if from_bus is None or to_bus is None:
+            print("Conversion of " + tline.name + " failed")
+            return False
+        network_frequency = core_model.base_frequency
+        # Calculate rated current
+        voltage = from_bus.nominal_voltage
+        rated_current = tline.rating/voltage
+        # Create line in pandapower network
+        pandapower.create_line_from_parameters(
+            net=self.network,
+            name=tline.name,
+            index=tline.uid,
+            geodata=tline.coords,
+            from_bus=from_bus.uid,
+            to_bus=to_bus.uid,
+            length_km=tline.length,
+            r_ohm_per_km=tline.r1,
+            x_ohm_per_km=tline.x1,
+            c_nf_per_km=(tline.b1 * 1e3) / (2 * pi * network_frequency),
```

Comment:
This calculation could also be a method of `TLine`. No need to change it for now, though. We might need to come up with a consistent plan for such cases.